### PR TITLE
BACKLOG-21626: Update label/description and check vanity action write…

### PR DESCRIPTION
**Pull request by jahia-carp**
https://jira.jahia.org/browse/BACKLOG-21626

## Summary by CodeRabbit

- **New Features**
	- Enhanced the Vanity URL feature to be enabled only when certain conditions are met.
- **Bug Fixes**
	- Updated user permissions to restrict access to the Vanity URL modal in read-only mode.
- **Documentation**
	- Updated labels and descriptions in multiple languages to reflect changes in SEO and Vanity URL features.
- **Tests**
	- Modified test cases to verify the updated user permissions and accessibility of the Vanity URL feature.

### CODE REVIEW

File: src/javascript/components/actions/VanityAction.jsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const VanityAction = ({render: Render, loading: Loading, label, requiredP
     return (
         <>
             <Render
-                enabled={mode !== 'create'}
+                enabled={mode !== 'create' && requiredPermission && nodeData.hasWritePermission}
```
Comment (dgriffon) [inline src/javascript/components/actions/VanityAction.jsx:38]: why do you need `&& requiredPermission` has it is always set?

**Review by dgriffon — APPROVED** [review]

File: tests/cypress/e2e/permissions/checkUisPermissions.cy.ts [hunk 2]
```diff
@@ -103,15 +103,15 @@ describe('Test UIs permissions', () => {
...
 
-    it('Verify that user have access to vanity url in content editor in read only', function () {
+    it('Verify that user have no access to vanity url modal in content editor in read only', function () {
         cy.login('secondEditorUser', 'password')
         const jcontent = JContent.visit(siteKey, 'en', 'pages/home')
         jcontent.switchToListMode()
         jcontent.editComponentByText(`${pageName}-a`)
         const contenteditor = new ContentEditorSEO()
+        contenteditor.checkVanityUrlAccessibility('false')
         const vanityUrlUi = contenteditor.openVanityUrlUi()
```
Comment (dgriffon) [inline tests/cypress/e2e/permissions/checkUisPermissions.cy.ts:113]: If the user has no access to the UI, what to we expect of such call?

**Review by dgriffon — DISMISSED** [review]

File: tests/cypress/e2e/permissions/checkUisPermissions.cy.ts (hunk 2 above)
Comment (jahia-carp) [reply tests/cypress/e2e/permissions/checkUisPermissions.cy.ts:113]: To verify that the UI shows a disabled button as expected now.

File: src/javascript/components/actions/VanityAction.jsx (hunk 1 above)
Comment (jahia-carp) [reply src/javascript/components/actions/VanityAction.jsx]: You are right it is not it is not necessary here, thanks.

File: tests/cypress/e2e/permissions/checkUisPermissions.cy.ts [hunk 3]
```diff
@@ -103,15 +103,15 @@ describe('Test UIs permissions', () => {
...
         vanityUrlUi.findReadOnlyBadge().should('exist')
```
Comment (jsinovassin) [inline tests/cypress/e2e/permissions/checkUisPermissions.cy.ts:114]: This check should not exists, it will always fail as the modal will never be open

**Review by jsinovassin — CHANGES_REQUESTED** [review]

File: tests/cypress/e2e/permissions/checkUisPermissions.cy.ts (hunk 3 above)
Comment (jahia-carp) [reply tests/cypress/e2e/permissions/checkUisPermissions.cy.ts:114]: This check allow us to verify that the "Manage Vanity Urls" entry in the 3dots menu is enabled or disabled, may be the name of the method is not rigth.

Comment (jahia-carp) [reply tests/cypress/e2e/permissions/checkUisPermissions.cy.ts]: updated, it's cleaner like that thanks

**Review by jsinovassin — APPROVED** [review]

**Pull request merged**